### PR TITLE
Enable kube-dns-autoscaler test in scale-correctness tests

### PR DESCRIPTION
**Pull request by wojtek-t**
```release-note
NONE
```

/sig scalability
/sig network
/kind cleanup
/priority important-longterm

### CODE REVIEW

File: test/e2e/autoscaling/dns_autoscaling.go [hunk 1]
```diff
@@ -166,8 +166,7 @@ var _ = SIGDescribe("DNS horizontal autoscaling", func() {
 		framework.ExpectNoError(err)
 	})
 
-	// TODO: Get rid of [DisabledForLargeClusters] tag when issue #55779 is fixed.
-	ginkgo.It("[DisabledForLargeClusters] kube-dns-autoscaler should scale kube-dns pods in both nonfaulty and faulty scenarios", func() {
+	ginkgo.It("kube-dns-autoscaler should scale kube-dns pods in both nonfaulty and faulty scenarios", func() {
```
Comment (prameshj) [inline test/e2e/autoscaling/dns_autoscaling.go:169]: Since https://github.com/kubernetes/kubernetes/issues/55779 is fixed, this is now safe to enable?

Comment (wojtek-t) [reply test/e2e/autoscaling/dns_autoscaling.go:169]: I think the fact that it's fixed doesn't mean anything.
But with endpoint-slices and couple other improvements that happened in the meantime it should be safe to enable it after those 3 years...

**Pull request merged**